### PR TITLE
feat-dynamic-deeplinks-support

### PR DESCRIPTION
**Pull request by IloveJavaa**
Added code for dynamic deeplinks
Asana task - https://app.asana.com/0/0/1206702369644644/f

### CODE REVIEW

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/APIRepository.kt [hunk 1]
```diff
@@ -41,13 +52,20 @@ object APIRepository {
         logger.info("Session body is: $body")
         return trackierApi.sendSessionData(body)
     }
+    
+    private suspend fun sendDeeplinks(body: MutableMap<String, Any>): ResponseData {
+        val logger = Factory.logger
+        logger.info("Deeplinks body is: $body")
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/APIRepository.kt:58]: Remove this

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt [hunk 2]
```diff
@@ -52,14 +55,31 @@ object TrackierSDK {
     }
 
     @JvmStatic
-    fun parseDeepLink(uri: Uri, context: Context) {
+    fun parseDeepLink(uri: Uri?, context: Context) {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:58]: Uri can not be null

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/ResponseData.kt [hunk 3]
```diff
@@ -24,6 +25,13 @@ data class ResponseData(
...
-) {
-}
\ No newline at end of file
+    val isRetargeting: Boolean?,
+    val message: String?,
+    val data: Data?
+)
+
+@JsonClass(generateAdapter = true)
+data class Data(
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/ResponseData.kt:34]: Let's rename it to DlData class

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt [hunk 4]
```diff
@@ -52,14 +55,31 @@ object TrackierSDK {
...
+        uri ?: return
         val ctx = context.applicationContext
         val prefs = Util.getSharedPref(ctx)
-        prefs.edit().putString(Constants.SHARED_PREF_DEEP_LINK, uri.query)
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:58]: Yeh sara code instance waali class mein kove kardo

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 5]
```diff
@@ -338,6 +350,21 @@ class TrackierSDKInstance {
         dlt.onDeepLinking(dlResult)
     }
     
+    fun callDeepLinkListenerDynamic(dlObj: ResponseData ) {
+        val dlt = this.config.getDeepLinkListener() ?: return
+        val dlResult: DeepLink
+        if (dlObj.data?.url == null){
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:356]: Again it can not be null

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/ResponseData.kt (hunk 3 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/ResponseData.kt:34]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt (hunk 4 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:58]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt (hunk 2 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt (hunk 5 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/APIRepository.kt (hunk 1 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/APIRepository.kt]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt [hunk 6]
```diff
@@ -4,6 +4,10 @@ import android.content.Context
 import android.net.Uri
 import android.util.Log
 import androidx.annotation.Keep
+import kotlinx.coroutines.CoroutineScope
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:7]: remove these imports as no longer used

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt [hunk 7]
```diff
@@ -52,15 +56,14 @@ object TrackierSDK {
...
-    fun parseDeepLink(uri: Uri, context: Context) {
-        val ctx = context.applicationContext
-        val prefs = Util.getSharedPref(ctx)
-        prefs.edit().putString(Constants.SHARED_PREF_DEEP_LINK, uri.query)
-            .remove(Constants.SHARED_PREF_DEEP_LINK_CALLED).apply()
-
-        if (instance.isInitialized) {
-            instance.callDeepLinkListener()
+    fun parseDeepLink(uri: Uri?) {
+        uri ?: return
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:60]: plz use if statement for return

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt [hunk 8]
```diff
@@ -338,10 +351,42 @@ class TrackierSDKInstance {
...
+    }
+    
     fun getRetargetingData(): MutableMap<String, Any> {
         val body = mutableMapOf<String, Any>()
         body["rtgtime"] = Util.getSharedPrefString(this.config.context, Constants.STORE_RETARGETING_TIME)
         body["url"] = Util.getSharedPrefString(this.config.context, Constants.STORE_RETARGETING)
         return body
     }
+    
+    fun parseDeepLink(uri: Uri) {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:374]: use uri: Uri?

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt (hunk 6 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:7]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt (hunk 8 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDKInstance.kt:374]: done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt (hunk 7 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/TrackierSDK.kt:60]: done

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**